### PR TITLE
Add support for plain table factories

### PR DESCRIPTION
**Pull request by ekmartin**
See https://github.com/facebook/rocksdb/wiki/PlainTable-Format.

### CODE REVIEW

File: src/db_options.rs [hunk 1]
```diff
@@ -982,6 +982,33 @@ impl Options {
...
+    /// let factory_opts = PlainTableFactoryOptions {
+    ///   user_key_length: 0,
+    ///   bloom_bits_per_key: 20,
+    ///   hash_table_ratio: 0.75,
+    ///   index_sparseness: 16,
+    /// };
+    ///
+    /// opts.set_plain_table_factory(factory_opts);
+    /// ```
+    pub fn set_plain_table_factory(&mut self, options: PlainTableFactoryOptions) {
```
Comment (iSynaptic) [inline src/db_options.rs]: Could the `options` be passed by immutable borrow?

**Comment by ekmartin** [comment]
@iSynaptic: changed it to an immutable borrow. I'm not sure what a unit test would add here—wouldn't it look identical to the existing doctest?

**Comment by iSynaptic** [comment]
@ekmartin Doc tests are not run as part of the CI build.

**Comment by ekmartin** [comment]
@iSynaptic: https://travis-ci.org/rust-rocksdb/rust-rocksdb/jobs/488739052#L461

**Comment by iSynaptic** [comment]
> 
> 
> @ekmartin Doc tests are not run as part of the CI build.

I've seen them not run in the past, but I just did an experiment and it did run.

**Comment by iSynaptic** [comment]
Ignore the build failure. That may be a result of my experiment. I'll merge and follow it through the `master` build.

**Pull request merged**